### PR TITLE
feat: publish Connect to npm on ci

### PR DESCRIPTION
**Pull request by acaldas**

### CODE REVIEW

File: studio/index.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            open: true,
+        },
+        plugins: [
+            viteConnectDevStudioPlugin(),
+            viteEnvs({
+                declarationFile: join(studioDirname, '../.env'),
+                computedEnv: studioConfig,
+            }),
+            runShellScriptPlugin(viteEnvsScript),
+            // WatchLocalPlugin(LOCAL_DOCUMENT_MODELS, LOCAL_DOCUMENT_EDITORS),
```
Comment (acaldas) [inline studio/index.ts:141]: Need to check if this watcher plugin is actually needed.

File: studio/index.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        customLogger: logger,
+        configFile: false,
+        root: appPath,
+        server: {
+            port: PORT,
+            open: true,
+        },
+        plugins: [
+            viteConnectDevStudioPlugin(),
+            viteEnvs({
```
Comment (acaldas) [inline studio/index.ts:136]: viteEnvs is used to set the environment variables inside the connect bundle.

File: studio/vite-plugin.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import path from 'node:path';
```
Comment (acaldas) [inline studio/vite-plugin.ts:1]: This vite plugin basically reads LOCAL_DOCUMENT_MODELS_IMPORT and LOCAL_DOCUMENT_EDITORS_IMPORT, resolves them into a filesystem path and sets in `resolve.alias`.

On the code we have a conditional `import("LOCAL_DOCUMENT_MODELS")` which resolves to that path. Vite then works it's magic and loads the package from that path, even if it's on typescript.

File: studio/vite-plugin.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+            if (resolvedAlias) {
+                resolve.alias = resolvedAlias;
+                config.resolve = resolve;
+            }
+        },
+        resolveId: id => {
+            // if the path was not provided then declares the local
+            // imports as external so that vite ignores them
+            if (importKeys.includes(id)) {
```
Comment (acaldas) [inline studio/vite-plugin.ts:97]: This is needed to make the server work if no LOCAL_DOCUMENT_MODELS are set, it makes vite ignore the import.

File: vite.renderer.config.mts [hunk 5]
```diff
@@ -38,6 +38,7 @@ export default defineConfig(({ mode }) => {
     const uploadSentrySourcemaps = authToken && org && project;
 
     const plugins: PluginOption[] = [
+        viteConnectDevStudioPlugin(env),
```
Comment (acaldas) [inline vite.renderer.config.mts:41]: `LOCAL_DOCUMENT_MODELS=./xxx npm run dev:web` is also supported!

File: src/store/document-model.ts [hunk 6]
```diff
@@ -2,35 +2,95 @@ import * as DocumentModels from 'document-model-libs/document-models';
...
+export const LOCAL_DOCUMENT_MODELS = import.meta.env.LOCAL_DOCUMENT_MODELS;
 
-export const documentModelsAtom = atom(documentModels);
+async function loadDynamicModels() {
+    if (!LOCAL_DOCUMENT_MODELS) {
+        return [];
+    }
+    try {
+        const localModules = (await import(
+            'LOCAL_DOCUMENT_MODELS'
```
Comment (acaldas) [inline src/store/document-model.ts:17]: This string will be resolved by vite to the filesystem path provided.

**Review by gpuente — APPROVED** [review]
Amazing! 🚀🚀🚀

**Pull request merged**